src/sdx_pce/topology/temanager.py

```diff
@@ -884,7 +884,7 @@ def _reserve_vlan(self, domain: str, port: dict, request_id: str, tag=None):
                 if vlan_usage is UNUSED_VLAN:
                     available_tag = vlan_tag
         else:
-            if vlan_table[tag] is UNUSED_VLAN:
+            if tag in vlan_table and vlan_table[tag] is UNUSED_VLAN:
```

Comment:
Issue https://github.com/atlanticwave-sdx/pce/issues/208 created to better discuss the behavior here